Chocolatey.Cake.Recipe/Content/psscriptanalyzer.cake

```diff
@@ -0,0 +1,123 @@
+// Copyright © 2023 Chocolatey Software, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+BuildParameters.Tasks.PSScriptAnalyzerTask = Task("Run-PSScriptAnalyzer")
+    .WithCriteria(() => BuildParameters.ShouldRunPSScriptAnalyzer, "Skipping because PSScriptAnalyzer is not enabled")
+    .WithCriteria(() => BuildParameters.ShouldRunAnalyze, "Skipping because running analysis tasks is not enabled")
+    .Does(() => RequirePSModule("PSScriptAnalyzer", "1.21.0", () =>
```

Comment:
Does this work on non-Windows, and does it provide a helpful error if PowerShell is not installed?
What happens on Windows if an older version of PowerShell is installed, or if pwsh is installed?
What happens if a different version of the module is already installed?

Reply:
These are all great questions! :-)

In this first pass, this will _only_ run on a Windows Machine.  It has been tested against Windows PowerShell, and it is known to work there.  Our build agents currently all use Windows PowerShell, so that is the only requirement here. This may well run when using pwsh, but at this stage, that would just be a bonus.  If a different version is already installed, it will force the installation of the required version.